src/Aliencube.AzureFunctions.Extensions.OpenApi.Core/Visitors/UriTypeVisitor.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using Aliencube.AzureFunctions.Extensions.OpenApi.Core.Abstractions;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json.Serialization;
+
+namespace Aliencube.AzureFunctions.Extensions.OpenApi.Core.Visitors
+{
+    /// <summary>
+    /// This represents the type visitor for <see cref="Uri"/>.
+    /// </summary>
+    public class UriTypeVisitor : TypeVisitor
+    {
+        /// <inheritdoc />
+        public UriTypeVisitor(VisitorCollection visitorCollection)
+            : base(visitorCollection)
+        {
+        }
+
+        /// <inheritdoc />
+        public override bool IsVisitable(Type type)
+        {
+            return type.IsAssignableFrom(typeof(Uri));
+        }
+
+        /// <inheritdoc />
+        public override void Visit(IAcceptor acceptor, KeyValuePair<string, Type> type, NamingStrategy namingStrategy, params Attribute[] attributes)
+        {
+            this.Visit(acceptor, name: type.Key, title: null, dataType: "string", dataFormat: "uri", attributes: attributes);
```

Comment:
By the way, I can't find the URI data type has the data format of "uri" from the spec.

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#data-types

Can you point me where you find this spec?

Reply:
Like "uuid", it is an extension of the data type string, so not directly part of the spec.

The spec allows it per: "to support documentation needs, the format property is an open string-valued property, and can have any value. Formats such as 'email', 'uuid', and so on, MAY be used even though undefined by this specification. Types that are not accompanied by a format property follow the type definition in the JSON Schema. Tools that do not recognize a specific format MAY default back to the type alone, as if the format is not specified." - http://spec.openapis.org/oas/v3.0.3

It is also noted as an extension here:
https://swagger.io/docs/specification/data-models/data-types/#string

Reply:
Got it. Thanks!